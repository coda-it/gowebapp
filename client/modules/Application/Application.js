import React from 'react';
import AlertPanel from 'client/modules/AlertPanel';
import { Loader } from 'graphen';
class Application extends React.PureComponent {
    componentDidMount() {
        const { mount } = this.props;
        mount();
    }
    render() {
        const { children, isLoaded } = this.props;
        return (React.createElement("div", { className: "application" },
            isLoaded && children,
            !isLoaded && React.createElement(Loader, null),
            React.createElement(AlertPanel, null)));
    }
}
export default Application;
//# sourceMappingURL=Application.js.map