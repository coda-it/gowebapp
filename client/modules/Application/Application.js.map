{"version":3,"file":"Application.js","sourceRoot":"","sources":["Application.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAC1B,OAAO,UAAU,MAAM,2BAA2B,CAAC;AACnD,OAAO,EAAE,MAAM,EAAE,MAAM,SAAS,CAAC;AAQjC,MAAM,WAAY,SAAQ,KAAK,CAAC,aAAoB;IAClD,iBAAiB;QACf,MAAM,EAAE,KAAK,EAAE,GAAG,IAAI,CAAC,KAAK,CAAC;QAC7B,KAAK,EAAE,CAAC;IACV,CAAC;IAED,MAAM;QACJ,MAAM,EAAE,QAAQ,EAAE,QAAQ,EAAE,GAAG,IAAI,CAAC,KAAK,CAAC;QAE1C,OAAO,CACL,6BAAK,SAAS,EAAC,aAAa;YACzB,QAAQ,IAAI,QAAQ;YACpB,CAAC,QAAQ,IAAI,oBAAC,MAAM,OAAG;YACxB,oBAAC,UAAU,OAAG,CACV,CACP,CAAC;IACJ,CAAC;CACF;AAED,eAAe,WAAW,CAAC"}