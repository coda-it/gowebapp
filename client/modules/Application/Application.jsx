// @flow
import React from 'react';
import AlertPanel from 'client/modules/AlertPanel';
import { Loader } from 'graphen';

type Props = {
  isLoaded: boolean,
  children: React$Element<any>,
  mount: () => void,
};

class Application extends React.PureComponent<Props> {
  componentDidMount() {
    const { mount } = this.props;
    mount();
  }

  render() {
    const { children, isLoaded } = this.props;

    return (
      <div className="application">
        {isLoaded && children}
        {!isLoaded && <Loader />}
        <AlertPanel />
      </div>
    );
  }
}

export default Application;
