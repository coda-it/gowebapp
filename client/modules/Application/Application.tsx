import React, { useEffect } from 'react';
import AlertPanel from 'client/modules/AlertPanel/AlertPanel';
import { Loader } from 'graphen';

type Props = {
  isLoaded: boolean;
  children: React.ReactNode;
  mount: () => void;
};

function Application({ isLoaded, children, mount }: Props) {
  useEffect(() => {
    console.log('xxx:app-useeffect');
    mount();
  }, [mount]);

  return (
    <div className="application">
      {isLoaded && children}
      {!isLoaded && <Loader />}
      <AlertPanel />
    </div>
  );
}

export default Application;
