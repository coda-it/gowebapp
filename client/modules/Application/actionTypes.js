export const MOUNT = 'application/mount';
export const LOADED = 'application/loaded';
//# sourceMappingURL=actionTypes.js.map