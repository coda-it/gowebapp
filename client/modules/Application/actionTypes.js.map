{"version":3,"file":"actionTypes.js","sourceRoot":"","sources":["actionTypes.ts"],"names":[],"mappings":"AAAA,MAAM,CAAC,MAAM,KAAK,GAAG,mBAAmB,CAAC;AACzC,MAAM,CAAC,MAAM,MAAM,GAAG,oBAAoB,CAAC"}