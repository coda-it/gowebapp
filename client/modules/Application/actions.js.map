{"version":3,"file":"actions.js","sourceRoot":"","sources":["actions.ts"],"names":[],"mappings":"AAAA,OAAO,KAAK,WAAW,MAAM,eAAe,CAAC;AAE7C,MAAM,CAAC,MAAM,KAAK,GAAG,GAAG,EAAE,CAAC,CAAC;IAC1B,IAAI,EAAE,WAAW,CAAC,KAAK;CACxB,CAAC,CAAC;AAEH,MAAM,CAAC,MAAM,MAAM,GAAG,GAAG,EAAE,CAAC,CAAC;IAC3B,IAAI,EAAE,WAAW,CAAC,MAAM;CACzB,CAAC,CAAC"}