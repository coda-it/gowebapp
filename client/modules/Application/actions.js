import * as actionTypes from './actionTypes';
export const mount = () => ({
    type: actionTypes.MOUNT,
});
export const loaded = () => ({
    type: actionTypes.LOADED,
});
//# sourceMappingURL=actions.js.map