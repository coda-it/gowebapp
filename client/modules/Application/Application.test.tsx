import _ from 'lodash';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Application from './Application';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  alerts: {
    alerts: [
      {
        type: 'info',
        message: 'Message 1',
        timestamp: new Date('2020-03-01'),
        isOld: false,
      },
    ],
  },
};
const store = mockStore(initialState);

describe('Application', () => {
  it('should render with loader', () => {
    const { container } = render(
      <Provider store={store}>
        <Application isLoaded={false} mount={_.noop}>
          Content
        </Application>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with children', () => {
    jest.useFakeTimers();

    const { container } = render(
      <Provider store={store}>
        <Application isLoaded mount={_.noop}>
          <p>Some content</p>
        </Application>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
