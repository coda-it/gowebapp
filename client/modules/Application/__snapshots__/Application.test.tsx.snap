// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render with children 1`] = `
<div>
  <div
    class="application"
  >
    <p>
      Some content
    </p>
    <div
      class="alert-panel"
    >
      <div
        class="alert-panel__alert alert-panel__alert--info  "
      >
        Message 1
      </div>
    </div>
  </div>
</div>
`;

exports[`Application should render with loader 1`] = `
<div>
  <div
    class="application"
  >
    <div
      class="gc-loader"
    />
    <div
      class="alert-panel"
    >
      <div
        class="alert-panel__alert alert-panel__alert--info  "
      >
        Message 1
      </div>
    </div>
  </div>
</div>
`;
